### DISCUSSION TITLE
Would you accept a PR to make cryptography/trustme optional for tests?

### DISCUSSION THREAD

**Discussion by mgorny** [Kludex/uvicorn#1939] [Ideas] [answered]
Hello,

On Gentoo we'd like to be able to run (a subset of) uvicorn's test suite on architectures where the Rust toolchain hasn't been ported yet (e.g. older ARM versions). Right now this is problematic because `conftest.py` imports `cryptography` and `trustme` packages unconditionally.

Would you accept a pull request that would make these imports optional? The usual idea would be to catch `ImportError`  and skip the relevant tests (I think the fixtures can cover most of them, if not all — but I'm open to other options at your preference.

**Comment by Kludex** [comment, marked as answer]
Sure. Whatever you need. 👍
